### PR TITLE
feat: add discussion reference to comments and responses

### PR DESCRIPTION
**Pull request by Ian2012**
It also adds the thread title to the thread events definition

**Description:** Describe in a couple of sentences what this PR adds

**JIRA:** Link to JIRA ticket

**Dependencies:** dependencies on other outstanding PRs, issues, etc.

**Merge deadline:** List merge deadline (if any)

**Installation instructions:** List any non-trivial installation
instructions.

**Testing instructions:**

1. Open page A
2. Do thing B
3. Expect C to happen
4. If D happened instead - check failed.

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Version bumped
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Create a tag
- [ ] Check new version is pushed to PyPI after tag-triggered build is
      finished.

**Author concerns:** List any concerns about this PR - inelegant
solutions, hacks, quick-and-dirty implementations, concerns about
migrations, etc.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @Ian2012! Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: event_routing_backends/processors/xapi/event_transformers/forum_events.py [hunk 1]
```diff
@@ -27,6 +27,11 @@ def get_object(self):
         object_id = self.get_data('data.id', True)
         object_path = self.get_data('context.path', True).rstrip('/').replace(object_id, '').rstrip('/')
 
+        kwargs = {}
+
+        if self.get_data('data.title'):
```
Comment (Ian2012) [inline event_routing_backends/processors/xapi/event_transformers/forum_events.py:32]: Only threads have a title.

File: event_routing_backends/processors/xapi/event_transformers/forum_events.py [hunk 2]
```diff
@@ -35,9 +40,31 @@ def get_object(self):
...
         )
 
+    def get_context_activities(self):
+        context_activities = super().get_context_activities()
+
+        discussion = self.get_data('data.discussion.id')
+        if not discussion:
+            return context_activities
+
+        context_activities.grouping = [
```
Comment (Ian2012) [inline event_routing_backends/processors/xapi/event_transformers/forum_events.py:54]: I'm using the grouping but we can also append it to the parent (which is the current course)

Comment (bmtcril) [reply event_routing_backends/processors/xapi/event_transformers/forum_events.py:54]: I think that "grouping" is good 👍

**Review by ziafazal — APPROVED** [review]
👍

**Comment by bmtcril** [comment]
@Ian2012 don't forget to bump version and such!

**Comment by openedx-webhooks** [comment]
@Ian2012 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/event-routing-backends&entry.867055334=https://github.com/openedx/event-routing-backends/pull/421&entry.1484655318=https://github.com/Ian2012&entry.752974735=2024-05-16+22:58&entry.1917517419=2024-05-22+15:16&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**